File: repos_estudiantes_TARQUI_primera_entrega/Quispe_Mollisaca_Jimmy_Brandon/Polimorfismo/Ejer1/Videojuego.java
package Polimorfismo;

public class Videojuego {
	
	private
	String nombre;
	String plataforma;
	int cantidadJugadores;
	

	public Videojuego(String nombre, String plataforma, int cantidadJugadores) {
		this.nombre = nombre;
		this.plataforma = plataforma;
		this.cantidadJugadores = cantidadJugadores;
	}
	
	public Videojuego(String nombre, String plataforma) {
		this.nombre = nombre;
		this.plataforma = plataforma;
		this.cantidadJugadores = 1;
	}
	
	
	public Videojuego() {
		this.nombre = "Roquet League";
		this.plataforma = "PlayStation";
		this.cantidadJugadores = 1;
	}
	
	public void mostrar() {
		System.out.println("Videojuego [nombre=" + nombre + ", plataforma=" + plataforma + ", cantidadJugadores="
				+ cantidadJugadores + "]");
	}
	
	public void agregarJugadores() {
	        cantidadJugadores += 1;
	        System.out.println("Se agregó 1 jugador. Total ahora: " + cantidadJugadores);
	}
	public void agregarJugadores(int nuevosJugadores) {
	        cantidadJugadores += nuevosJugadores;
	        System.out.println("Se agregaron " + nuevosJugadores + " jugadores. Total ahora: " + cantidadJugadores);
	}

	public static void main(String[] args) {
		Videojuego v1 = new Videojuego("Minecraft", "PC", 5);
        Videojuego v2 = new Videojuego("FIFA 23", "PlayStation");

        System.out.println("--- Videojuego 1 ---");
        v1.mostrar();
        System.out.println("--- Videojuego 2 ---");
        v2.mostrar();

        v1.agregarJugadores();
        v1.agregarJugadores(3);

        v2.agregarJugadores();
        v2.agregarJugadores(4);
	}

}

File: repos_estudiantes_TARQUI_primera_entrega/Quispe_Mollisaca_Jimmy_Brandon/Polimorfismo/Ejer1/Videojuego.py
class Videojuego:
    def __init__(self, nombre="Rocket League", plataforma="PlayStation", cantidadJugadores=1):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cantidadJugadores = cantidadJugadores

    def mostrar(self):
        print(f"Videojuego [nombre={self.nombre}, plataforma={self.plataforma}, cantidadJugadores={self.cantidadJugadores}]")

    def agregarJugadores(self):
        self.cantidadJugadores += 1
        print(f"Se agregó 1 jugador. Total ahora: {self.cantidadJugadores}")

    def agregarJugadoresCantidad(self, nuevosJugadores):
        self.cantidadJugadores += nuevosJugadores
        print(f"Se agregaron {nuevosJugadores} jugadores. Total ahora: {self.cantidadJugadores}")


v1 = Videojuego("Minecraft", "PC", 5)
v2 = Videojuego("FIFA 23", "PlayStation")

print("--- Videojuego 1 ---")
v1.mostrar()
print("--- Videojuego 2 ---")
v2.mostrar()

v1.agregarJugadores()
v1.agregarJugadoresCantidad(3)

v2.agregarJugadores()
v2.agregarJugadoresCantidad(4)